composites/Plugin/SnippetEditor/components/SnippetEditor.js

```diff
@@ -414,34 +415,58 @@ class SnippetEditor extends React.Component {
 	 * @returns {Object} The data for the preview.
 	 */
 	mapDataToMeasurements( originalData, replacementVariables = this.props.replacementVariables ) {
-		const { baseUrl, mapEditorDataToPreview } = this.props;
+		const { baseUrl, permalink = baseUrl, mapEditorDataToPreview } = this.props;
 
 		let description = this.processReplacementVariables( originalData.description, replacementVariables );
 
 		// Strip multiple spaces and spaces at the beginning and end.
 		description = stripSpaces( description );
 
-		const shortenedBaseUrl = baseUrl.replace( /^http:\/\//i, "" );
+		const shortenedBaseUrl   = baseUrl.replace( /^http:\/\//i, "" );
+		const shortenedPermalink = permalink.replace( /^http:\/\//i, "" );
 
 		const mappedData = {
 			title: this.processReplacementVariables( originalData.title, replacementVariables ),
-			url: shortenedBaseUrl + originalData.slug,
+			url: this.buildPermalink( shortenedPermalink, originalData ),
 			description: description,
 		};
 
-		const context = {
-			shortenedBaseUrl,
-		};
-
 		// The mapping by the passed mapping function should happen before measuring.
 		if ( mapEditorDataToPreview ) {
-			return mapEditorDataToPreview( mappedData, context );
+			return mapEditorDataToPreview(
+				mappedData,
+				{ shortenedBaseUrl }
+			);
 		}
 
-
 		return mappedData;
 	}
 
+	/**
+	 * Builds the permalink based on the permalink structure and the passed URL.
+	 *
+	 * If categories are a part of the permalink, then this will be replaced as well.
+	 *
+	 * @param {string} url          The URL to base the permalink on.
+	 * @param {Object} originalData The original post data.
+	 *
+	 * @returns {string} The formatted permalink.
+	 */
+	buildPermalink( url, originalData ) {
+		let parents = [];
+		const { slug, primaryTaxonomySlug = "" } = originalData;
+
+		if ( originalData.parents ) {
+			parents = Object.values( originalData.parents );
+		}
+
+		const categorySlug = [ ...parents, primaryTaxonomySlug ].join( "/" );
+		let permalink      = url.replace( "%category%", categorySlug );
```

Comment:
This is WordPress specific. Is there a way to move this to the Yoast SEO plugin?